include/core/ib_particle.h

```diff
@@ -89,12 +98,95 @@ class IBParticle
   unsigned int
   get_number_properties();
 
+  /**
+   * @brief
+   * Returns the evaluation of the signed distance function of this solid at the
+   * given point p
+   * Most levelset functions come from Inigo Quilez:
+   * iquilezles.org/articles/distfunctions
+   */
+  double
+  get_levelset(const Point<dim> &p);
+
+  /**
+   * @brief
+   * Sets the closest_point parameter to be the point on the surface of the
+   * solid which has the minimal distance from the given point p
+   */
+  void
+  closest_surface_point(const Point<dim> &p, Point<dim> &closest_point);
+
+  /**
+   * @brief
+   * Returns true if the given point is inside the crown for which the limits
+   * are defined by inner and outer radius factors. An effective radius is used
+   * for non spherical shapes.
+   */
+  bool
+  is_inside_crown(const Point<dim> &evaluation_pt,
+                  const double      outer_radius,
+                  const double      inside_radius);
+
+  /**
+   * @brief
+   * Sets up a default shape
+   */
+  void
+  initialize_shape();
+
+  /**
+   * @brief
+   * Sets up a shape in accordance with the given type and arguments
+   */
+  void
+  initialize_shape(const std::string type, Tensor<1, 3> solid_arguments);
+
+  /**
+   * @brief
+   * Sets the position of the particle and dependent members to the argument
+   */
+  void
+  set_position(const Point<dim> position);
+
+  /**
+   * @brief
+   * Sets the position of the particle and dependent members to the argument for
+   * one component
+   */
+  void
+  set_position(const double       position_component,
+               const unsigned int component = 0);
+
+  /**
+   * @brief
+   * Sets the orientation of the particle and dependent members to the argument
+   */
+  void
+  set_orientation(const Tensor<1, 3> orientation);
+
+  /**
+   * @brief
+   * Sets the center of rotation offset of the particle and dependent members to
+   * the argument
+   */
+  void
+  set_center_of_rotation_offset(const Point<dim> cor_offset);
+
+
   // This class defines values related to a particle used in the sharp interface
   // IB. Each particle defined will have these value used in the solver.
 
   // The unique identification number of the particle.
   unsigned int particle_id;
-  // The particle radius.
+
+  // The geometrical information regarding the particle
+  std::shared_ptr<Shape<dim>> shape;
+  // The function from which the solid arguments are parsed
+  std::shared_ptr<Functions::ParsedFunction<3>> f_solid_arguments;
```

Comment:
I don't understand this part. Why do you need a function for the solid arguments? Why not just an std::vector<string> built from the argument.